addons/udes_stock_refactoring/models/stock_move.py

```diff
@@ -155,7 +155,13 @@ def _action_assign(self):
         res = super(StockMove, self)._action_assign()
 
         refactored_moves = res._action_refactor(stage="assign")
-        res |= refactored_moves
+
+        # Merge possible moves (if same group/location/destination) and their move lines
+        # Note that _merge_moves() may return same or different move(s) after merging other moves (or no merging)
+        refactored_moves = refactored_moves._merge_moves()
```

Comment:
Can we say that merging moves is a process which is related with refactoring rather than assigning? 
My suggestion in that case maybe to move the _merge_moves and _merge_move_lines inside _action_refactor.

Reply:
Makes sense. I have moved the merging into `_action_refactor` and tested to be working well.

One question: should the moves be merged only if `new_moves != stage_moves`? Can it happen that even if refactoring didn't change/create moves, there are other possible moves that could be merged?

CC: @ws-mps

Reply:
Sounds reasonable to move it. Moves do not necessarily will change, they might get move into another picking only. I Wouldn't expect to have other moves that could be merged, since everything should get merged when it was refactored in the first place.

Reply:
Ok, IIUC, even if refactor doesn't change the set of moves; those moves could be potentially merged; so we proceed to merge (as is).